provider/datagen/src/driver.rs

```diff
@@ -600,73 +889,81 @@ fn select_locales_for_key(
         });
     }
 
-    let result = match (explicit_langids, fallback) {
-        // Case 1: `None` simply exports all supported locales for this key.
-        (None, _) => supported_map.into_values().flatten().collect(),
+    if locales_fallback.is_all_locales() {
+        // Case 1: `is_all_locales` simply exports all supported locales for this key.
+        return Ok(supported_map.into_values().flatten().collect());
+    }
```

Comment:
thought: this is a special case of case 3 now, so I'd put it in there. It's also purely an optimisation, we'd still get the same result without it. I would remove the `is_all_locales` method and do a check on the list of locales directly here. Also it suffices to check that the list contains `und`, it doesn't have to be the only member.